### PR TITLE
Put everything in paddle python wheel package enable pip install paddle

### LINKED ISSUE #2961: 提供可以pip安装的python包
提个问题：外部安装的时候能不能使用pip直接安装，好多人习惯了pip install ***的模式。最大可能的简化安装过程，降低学习和使用者的门槛，有利于产品的推广。。

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix #2961

So we can push this paddle wheel package to pypi for users to simple run `pip install paddle` to install

### CODE REVIEW

**Comment by helinwang** [comment]
Amazing!!!

**Comment by typhoonzero** [comment]
@helinwang CI fixed. And if we are going to make pushing `whl` packages to `pypi.python.org` by ci, how can we seperate cpu and gpu versions, and also seperate binaries for Linux and MacOS?

File: python/setup.py.in [hunk 1]
```diff
@@ -31,13 +32,22 @@ setup(name='paddle',
...
       package_dir={
           '': '${CMAKE_CURRENT_SOURCE_DIR}',
           # The paddle.v2.framework.proto will be generated while compiling.
           # So that package points to other directory.
-          'paddle.v2.framework.proto': '${PROJ_BINARY_ROOT}/paddle/framework'
+          'paddle.v2.framework.proto': '${PROJ_BINARY_ROOT}/paddle/framework',
+          'py_paddle': '${PROJ_ROOT}/paddle/py_paddle'
       },
+      scripts=['${PROJ_BINARY_ROOT}/paddle/scripts/paddle'],
+      data_files=[('/usr/local/opt/paddle/bin',
```
Comment (gangliao) [inline python/setup.py.in]: Dumping all binaries into wheel package will make it too large. Why we need this?

Comment (typhoonzero) [inline python/setup.py.in]: Then people can just run `pip install paddle` to install paddle.

**Comment by helinwang** [comment]
@typhoonzero did some search. For different binary for linux and mac, seems we need to build "platform wheel" (run `python setup.py bdist_wheel` on every target platforms): https://packaging.python.org/tutorials/distributing-packages/#platform-wheels

I think we need to have different pip package name for GPU and CPU only. E.g., tensorflow uses:
```text
 $ pip install tensorflow      # Python 2.7; CPU support (no GPU support)
 $ pip install tensorflow-gpu  # Python 2.7;  GPU support
```

**Pull request merged**